sorting/merge_sort.c

```diff
@@ -1,139 +1,89 @@
 /**
- * @file
- * @brief Implementation of [merge
- * sort](https://en.wikipedia.org/wiki/Merge_sort) algorithm
+ *@author [Bama Charan Chhandogi](https://github.com/BamaCharanChhandogi)
  */
 #include <stdio.h>
-#include <stdlib.h>
 
-/**
- * @addtogroup sorting Sorting algorithms
- * @{
- */
-/** Swap two integer variables
- * @param [in,out] a pointer to first variable
- * @param [in,out] b pointer to second variable
- */
-void swap(int *a, int *b)
+// Function to merge two sorted subarrays into one sorted array
+void merge(int arr[], int left, int mid, int right)
 {
-    int t;
-    t = *a;
-    *a = *b;
-    *b = t;
-}
+    int i, j, k;
+    int n1 = mid - left + 1;
+    int n2 = right - mid;
 
-/**
- * @brief Perform merge of segments.
- *
- * @param a array to sort
- * @param l left index for merge
- * @param r right index for merge
- * @param n total number of elements in the array
- */
-void merge(int *a, int l, int r, int n)
-{
-    int *b = (int *)malloc(n * sizeof(int)); /* dynamic memory must be freed */
-    if (b == NULL)
-    {
-        printf("Can't Malloc! Please try again.");
-        exit(EXIT_FAILURE);
-    }
-    int c = l;
-    int p1, p2;
-    p1 = l;
-    p2 = ((l + r) / 2) + 1;
-    while ((p1 < ((l + r) / 2) + 1) && (p2 < r + 1))
+    // Create temporary arrays to store the left and right subarrays
+    int L[n1], R[n2];
+
+    // Copy data to temporary arrays
+    for (i = 0; i < n1; i++) L[i] = arr[left + i];
+    for (j = 0; j < n2; j++) R[j] = arr[mid + 1 + j];
+
+    // Merge the temporary arrays back into arr[left..right]
+    i = 0;
+    j = 0;
+    k = left;
+    while (i < n1 && j < n2)
     {
-        if (a[p1] <= a[p2])
+        if (L[i] <= R[j])
         {
-            b[c++] = a[p1];
-            p1++;
+            arr[k] = L[i];
+            i++;
         }
         else
         {
-            b[c++] = a[p2];
-            p2++;
+            arr[k] = R[j];
+            j++;
         }
+        k++;
     }
 
-    if (p2 == r + 1)
+    // Copy the remaining elements of L and R, if any
+    while (i < n1)
     {
-        while ((p1 < ((l + r) / 2) + 1))
-        {
-            b[c++] = a[p1];
-            p1++;
-        }
+        arr[k] = L[i];
+        i++;
+        k++;
     }
-    else
+    while (j < n2)
     {
-        while ((p2 < r + 1))
-        {
-            b[c++] = a[p2];
-            p2++;
-        }
+        arr[k] = R[j];
+        j++;
+        k++;
     }
-
-    for (c = l; c < r + 1; c++) a[c] = b[c];
-
-    free(b);
 }
 
-/** Merge sort algorithm implementation
- * @param a array to sort
- * @param n number of elements in the array
```

Comment:
this style of documentation was prefered

Reply:
I have added these portions.